src/adapterManager.js

```diff
@@ -173,21 +173,43 @@ function getAdUnitCopyForClientAdapters(adUnits) {
 
 export let gdprDataHandler = {
   consentData: null,
-  setConsentData: function(consentInfo) {
+  generatedTime: null,
+  setConsentData: function(consentInfo, time = timestamp()) {
     gdprDataHandler.consentData = consentInfo;
+    gdprDataHandler.generatedTime = time;
   },
   getConsentData: function() {
     return gdprDataHandler.consentData;
+  },
+  getConsentMeta: function() {
+    if (gdprDataHandler.consentData && gdprDataHandler.consentData.vendorData && gdprDataHandler.consentData.vendorData.tcString && gdprDataHandler.generatedTime) {
+      return {
+        gdprApplies: gdprDataHandler.consentData.gdprApplies,
+        consentStringSize: gdprDataHandler.consentData.vendorData.tcString.length,
```

Comment:
I don't know well the behavior related to the setting of `.vendorData.tcString` but I think we don't have to change the main condition. With your change, if the tcString is `""` the function will returns nothing, and it doesn't sound what you want;

In my first comment I thought something like the code below.

```js
if (gdprDataHandler.consentData && gdprDataHandler.consentData.vendorData && gdprDataHandler.generatedTime) {
      return {
        gdprApplies: gdprDataHandler.consentData.gdprApplies,
        consentStringSize: (isStr(gdprDataHandler.consentData.vendorData.tcString)) ? gdprDataHandler.consentData.vendorData.tcString.length : 0,
        …
```

Please let me know

Reply:
@osazos Agreed, that seems more elegant. Let me update.